app/components/search.tsx

```diff
@@ -31,6 +33,17 @@ export default function Search({limitFromUrl, className}: Props) {
     }
   }
 
+  useEffect(() => {
+    const query = searchParams.get('q')
+    if (loadedQuestions && query) {
+      searchFn(query)
+      setShowResults(true)
+    }
+    // Properly adding all dependancies would require transforming `searchFn` into a callback
```

Comment:
from phone, but something like:
```ts
const searchFnRef = useRef(searchFn)
searchFnRef.current = searchFn
useEffect(
...searchFnRef.current(),
...
```

Reply:
dunno if that's better... :D

Reply:
disabling the eslint rule is unsafe for future changes which can more easily [introduce bugs](https://react.dev/learn/removing-effect-dependencies#to-change-the-dependencies-change-the-code).. mutable refs might not look good, but you explicitly say "this is not a dependency which triggers a synchronization, I just want to use whatever value is has currently" without implying anything about any other variable that may or may not be a dependency